### DISCUSSION TITLE
Allow collection-specific field labels

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5601] [Feature Requests]
Field labels set in the schema apply across all collections. CSIRO would like for collection-specific field names to be possible in the schema configurator.

**Original Message**

> In both the ANFC and ANWC, a number of fields are used across multiple collections (e.g., catalognumber field, some date and agent fields), with the field Caption being different between the collections. Changes to the field name caption have been made in the forms via the form xml, however, in searches the Caption reverts to what is recorded in the schema and does not reflect the Captions for the individual collection. This is confusing for the end-user.
>  
> e.g., in the ANFC, the catalognumber is called ‘CSIRO Reg No’ for specimens, ‘Genetic tissue no’ for tissues, and ‘Plate no’ for xrays.
>  
> It would be good if the schema could include an option for Captions per collection.

[Related to ANFC #84]

**Requested By:** Corrina P at CSIRO

**Comment by maxpatiiuk** [comment]
This would be a bit complicated and not compatible with sp6

The best workaround would be to specify the label directly on the form definition. The form definition can be scoped to institution/discipline/collection/user type/user

See the "label" attribute - https://github.com/specify/specify7/wiki/Form-System#label

**Comment by grantfitzsimmons** [comment]
> This would be a bit complicated and not compatible with sp6
> 
> The best workaround would be to specify the label directly on the form definition. The form definition can be scoped to institution/discipline/collection/user type/user
> 
> See the "label" attribute - https://github.com/specify/specify7/wiki/Form-System#label

Corinna mentions that in her message to us, but unfortunately when you are creating a query or using any view outside of the form it causes issues.

**Comment by maxpatiiuk** [comment]
For back-end:
Need to extend the `/context/schema_localization.json?lang=en` endpoint to support passing `Collection.id`, or `SpLocalContainer.id`

For front-end:
Need to adapt the schema-editor for editing collection-specific field labels

**Comment by benanhalt** [comment]
The problem is they want different schema localizations for collections in the same discipline?

**Comment by grantfitzsimmons** [comment]
I think that's exactly it- they want different schema localizations for collections in the same discipline.

**Comment by benanhalt** [comment]
Could they just create extra disciplines?

**Comment by tlammer** [comment]
They could but then they couldn't share the data like they can with collections in the same discipline

**Comment by tlammer** [comment]
this is a really common request

**Comment by benanhalt** [comment]
I don't think it can happen until we break compatibility with 6.

**Comment by maxpatiiuk** [comment]
@benanhalt What if we define two English languages in the database (i.e, `en-gb` and `en-us`). Then, you can specify different labels for different languages (with fallback to `en` if no string is present)

Then, in Sp7 user preferences, set `en-gb` as language in one collection and `en-us` in another.

Thus, this hack already allows having different labels in different collections.

Though, it won't work in Sp6 as schema language pref in sp6 are scoped to computer, not collection
